### ISSUE TITLE
Extract a meaningful, non-empty string type and apply to the xsd schema

### ISSUE THREAD

**Issue by ykulbak** [buildingSMART/BCF-XML#235]
Many types in schema are not meaningful:

e.g.

```
    <xs:simpleType name="Stage">
        <xs:restriction base="xs:string"/>
    </xs:simpleType>
```

This change would expose a more meaningful type of `non-empty-string` and apply it in all appropriate places.

**Comment by ykulbak** [comment]
@CBenghi

**Comment by ykulbak** [comment]
@jasollien please review #248

**Comment by CBenghi** [comment]
Apologies for late response, I'm super stretched these days.

I've looked into this and it's a quick fix, but need an agreement with the group. 

A few doubts:

1.  Currently we have a `guid.xsd`, which could be renamed to `shared.sxd` and include the `NonEmptyString` type at this stage, and others that we might implement later. I've prepared some of these changes can create a PR quickly.
2.  While this does not affect the XML, it might have an impact for implementations: if any user auto-generate the code from the xsd, they will have an harder time to specialise these strings to any needs they might have (vocabulares and user UI specific constraints).

Since we have discussed this issue in the call without Klaus, I'm a bit uncomfortable with the changes.

**Comment by CBenghi** [comment]
@ykulbak we have posted contemporarily... I have the same code ready in my repo, but see point 2 in my previous code.

**Comment by ykulbak** [comment]
Discussed in the telecon today and we decided that @jasollien will review - but in general we're not concerned by point 2 and we think it's a good change.

**Comment by ykulbak** [comment]
Closed by #248
